packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -382,6 +382,10 @@ export abstract class BaseSegment extends MergeNode implements ISegment {
 	public localSeq?: number;
 	public localRemovedSeq?: number;
 
+	private get attributionI(): IAttributionCollectionInternal<AttributionKey> | undefined {
```

Comment:
to avoid casting i would change the typeing here to be something like `IAttributionCollection & Partial<IMergeTreeAttributionCollectionExtension>` the partial should allow implicit conversion from and to IAttributionCollection without casting, and since you don't have a way to know if the extension actually exists the partial keeps it type safe as they all need to be optional.

Reply:
i see the situation is a bit more complicated, but i still think we can get a better solution that doesn't require all the casting, especially as all the types and implementations are really owned by this package.